### DISCUSSION TITLE
베타ver 패키지 구조 수정 및 추가사항 정리

### DISCUSSION THREAD

**Discussion by silverpoodle** [meta-mingles/metamingle-server#132] [Troubleshooting]
### 패키지 구조 리팩토링
```
mingles
└─ metamingle
   ├─ avatar
   ├─ creativetool
   ├─ global
   │  ├─ auth
   │  ├─ common
   │  ├─ configuration
   │  └─ infra
   ├─ interactivemovie
   ├─ member
   ├─ movielike
   ├─ quiz
   ├─ scenario
   ├─ shortform
   └─ shortformlike
   ```
<br>
 **global 패키지**

*해당 패키지에는 특정 domain에 종속되지 않고, 프로젝트 전방위적으로 사용할 수 있는 클래스들이 모여있습니다.*
- auth : 인증, 인가와 관련된 클래스들로 구성
- common : 공통 클래스 혹은 공통 value 클래스들로 구성
- config : 각종 configuration 클래스로 구성
- infra : 외부 api 를 사용하는 클래스로 구성

#### AI 서버와 통신하는 기능은 모두 global/infra 패키지에 들어있습니다.

참고: https://velog.io/@jsb100800/Spring-boot-directory-package
<br>
<br>

#### 1. 등록 - http://localhost:8080/member/admin/signup (POST)
```json
{
    "email": "admin@gmail.com",
    "nickname": "관리자1",
    "password": "password"
}
```
<br>

#### 2. 로그인- http://localhost:8080/member/login(POST) 
```json
{
    "email": "admin@gmail.com",
    "password": "password"
}
```
- AI 서버 주소 변경을 위해 추가하였습니다.
- security 로 ADMIN 권한이 있는 회원만 접근가능하도록 설정하였습니다.

<br>
<br>

#### 1. 업데이트 - http://localhost:8080/infra/update-addr (POST)
```
 json 아니고 text 로 바로 전송하면 됩니다. (PostMan)
```

<br>

#### 2. 조회- http://localhost:8080/infra/get-addr(GET)
*주소 확인용 API*

@numerical43

**Comment by numerical43** [comment]
깔끔한 정리 감사합니다~😉
